cabot/metricsapp/api/grafana.py

```diff
@@ -0,0 +1,195 @@
+import json
+import logging
+import requests
+import urlparse
+from django.core.exceptions import ValidationError
+
+
+logger = logging.getLogger(__name__)
+
+
+def _get_time_range(timestring):
+    """
+    Find the number of digits a time string corresponds to.
+    :param timestring: time string in the format "3h" or "1d" or "10m"
+    :return: integer number of minutes
+    """
+    time_range = None
+    time_digits = int(filter(str.isdigit, timestring))
+
+    # Days
+    if 'd' in timestring:
+        time_range = 1440 * time_digits
+
+    # Hours
+    elif 'h' in timestring:
+        time_range = 60 * time_digits
+
+    # Minutes
+    elif 'm' in timestring:
+        time_range = time_digits
+
+    return time_range
+
+
+def get_grafana_auth_headers(api_key):
+    """
+    Returns request headers for accessing the Grafana API with an API key
+    :param api_key: the API key
+    :return: dictionary of relevant headers
+    """
+    return {'Authorization': 'Bearer {}'.format(api_key)}
+
+
+def get_dashboards(url, api_key):
+    """
+    Get data for all Grafana dashboards
+    :param url: url for the Grafana instance
+    :param api_key: API key for the Grafana instance
+    :return: api response containing dashboard id, title, uri, etc.
+    """
+    response = requests.get(urlparse.urljoin(url, 'api/search'), headers=get_grafana_auth_headers(api_key))
+
+    if response.status_code == 200:
+        return json.loads(response.text)
+    else:
+        logger.exception('Request to {} failed'.format(url))
+        raise ValidationError('Request to Grafana API failed.')
+
+
+def get_dashboard_choices(api_response):
+    """
+    Given a response from the Grafana API, find a list of dashboard name choices
+    :param api_response: response data from the Grafana API
+    :return: list of dashboard URIs and names
+    """
+    # Uri format is db/dashboard-name. Get rid of the leading db.
+    return [(dash['uri'].split('/')[1], dash['title']) for dash in api_response]
+
+
+def get_dashboard_info(grafana_url, api_key, dashboard_uri):
+    """
+    Get data about a Grafana dashboard
+    :param url: url for the Grafana instance
+    :param api_key: API key for the Grafana instance
+    :param dashboard_uri: dashboard part of the url
+    :return: api response containing creator information and panel information
+    """
+    http_api = urlparse.urljoin(grafana_url, 'api/dashboards/db/')
```

Comment:
You can use requests mounting to make this easier - http://docs.python-requests.org/en/master/user/advanced/#transport-adapters

So, you can set up a requests.Session with a prefix of grafana_url. Then you need to do
```
session.get('/api/dashboards/db/')
```

Never tried it before. YMMV